File: models/policy.ts
import mongoose, { Schema, Document, Model } from 'mongoose'
import type { PolicyStatus } from '@/types'

export interface IPolicy extends Document {
  _id: mongoose.Types.ObjectId
  userId: mongoose.Types.ObjectId
  planId: mongoose.Types.ObjectId
  status: PolicyStatus
  premium: number
  startDate: Date
  endDate: Date
  cropType: string
  landSize: number
  location: string
}

const PolicySchema = new Schema<IPolicy>(
  {
    userId: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    planId: {
      type: Schema.Types.ObjectId,
      ref: 'Plan',
      required: true,
    },
    status: {
      type: String,
      enum: ['active', 'expired', 'cancelled'],
      default: 'active',
    },
    premium: {
      type: Number,
      required: true,
    },
    startDate: {
      type: Date,
      required: true,
    },
    endDate: {
      type: Date,
      required: true,
    },
    cropType: {
      type: String,
      required: true,
    },
    landSize: {
      type: Number,
      required: true,
    },
    location: {
      type: String,
      required: true,
    },
  },
  {
    timestamps: true,
  }
)

const Policy: Model<IPolicy> = mongoose.models.Policy || mongoose.model<IPolicy>('Policy', PolicySchema)

export default Policy

File: backend/models/policy.py
from sqlalchemy import Column, String, Float, Integer, DateTime, ForeignKey, func
from backend.database import Base


class Policy(Base):
    __tablename__ = "policies"

    id = Column(String, primary_key=True)
    user_id = Column(String, ForeignKey("users.id"), nullable=False, index=True)
    plan_id = Column(String, ForeignKey("plans.id"), nullable=False)
    status = Column(String, default="active")  # active, expired, cancelled
    premium = Column(Float, nullable=False)
    start_date = Column(DateTime, nullable=False)
    end_date = Column(DateTime, nullable=False)
    crop_type = Column(String, nullable=False)
    land_size = Column(Float, nullable=False)
    location = Column(String, nullable=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
